pkg/sql/plan/build_dml_util.go

```diff
@@ -859,6 +860,7 @@ func makeInsertPlan(
 					return err
 				}
 
+				checkInsertPkDup = indexdef.Unique // only check PK dedup for UK. SK will not check PK dedup.
```

Comment:
you should not reuse this var, make a new one.
you want to set checkInsertPkDup's value for another makeInserPlan(),  not this context.